up-l1/mqtt_5.adoc

```diff
@@ -0,0 +1,276 @@
+
+[cols="1,1"]
+|===
+| MQTT Header Property Code | Value (key:value)
+| UserProperty
+| ("1", "{UAttribute.id}") # UUID id
+| UserProperty
+| ("2", "{UAttribute.type}") # UMessageType
+| UserProperty
+| ("3","{UAttribute.source}") # Short Form Serialized Source UURI
+| UserProperty
+| ("4","{UAttribute.sink}") # Short Form Serialized Sink UURI
+| UserProperty
+| ("5","{UAttribute.priority}") # UPriority
+| UserProperty
+| ("9", "{UAttribute.reqId}") # UUID reqid
+| UserProperty
+| ("opt_ua", "{byte array of remaining UAttributes}") # Remaining UAttributes
+|===
+
+All required uAttributes are mapped to a MQTT header UserProperty, where the key is the UAttribute protobuf field number.
+
+The source and sink fields **MUST** be a short form serialized UUri.
+
+Optional uAttributes will be a string representation of a byte array under the UserProperty field with key "opt_ua". This is done to reduce the size of the overall payload. Protobuf serialized byte arrays do not include values in a protobuf message object that are not set, so the byte array will only include the optional UAttributes that are set.
+
+=== URI Mapping to MQTT topics
+
+The MQTT topic a message is published on utilizes elements from both the source and sink UUri fields. Which elements from the UUri are used in the topic depend on whether the sink UUri is present.
+
+Has sink UUri:
+
+`{source.authority.domain}/{source.authority.name}/{sink.authority.domain + ":" + sink.authority.name}/{sink.entity.id}/{sink.entity.version_major}/{sink.resource.id}`
+
+No sink UUri (Publish message type only):
+
+`{source.authority.domain}/{source.authority.name}//{source.entity.id}/{source.entity.version_major}/{source.resource.id}`
+
+The entity and resource elements always represent the destination of the message. In the case of a Publish message type, the destination happens to be based on the source UUri. The source authority information is also always present to allow for authorization controls based on the authority information, since an authority maps to the uE sending the message. The domain information is included is separated out as well to allow for the cloud to easily subscribe to messages from devices only.
+
+Below are examples of messages being sent and how to subscribe to those messages:
+
+==== Example 1: Notification or RPC Message Types
+
+[source]
+----
+source:
+UUri {
+  authority: UAuthority {
+    name: "Vehicle_1",
+    domain: veh.com,
+  },
+  entity: UEntity {
+    version_major: 1,
+    id: 1234,
+  },
+  resource: UResource {
+    id: 5678,
+  }
+}
+
+sink:
+UUri {
+  authority: UAuthority {
+    name: "Vehicle_2",
+    domain: veh.com,
+  },
+  entity: UEntity {
+    version_major: 1,
+    id: 4321,
+  },
+  resource: UResource {
+    id: 8765,
+  }
+}
+----
+topic messages are published to: `veh.com/Vehicle_1/veh.com:Vehicle_2/4321/1/8765`
+
+to register a listener with this topic, the following source and sink filter would need to be provided:
```

Comment:
Use the short form string now to represent the URI

Reply:
Updated